slam-libraries/viam-cartographer/src/io/file_handler_test.cc

```diff
@@ -0,0 +1,183 @@
+#include "file_handler.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <boost/test/unit_test.hpp>
+#include <ctime>
+#include <exception>
+
+namespace viam {
+namespace io {
+namespace {
+
+BOOST_AUTO_TEST_SUITE(file_handler)
+
+BOOST_AUTO_TEST_CASE(MakeFilenameWithTimestamp_success) {
+    std::string path_to_dir = "path_to_dir";
+    std::time_t start_time =
+        std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    std::string filename = MakeFilenameWithTimestamp(path_to_dir);
+    std::time_t end_time =
+        std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    // Check if the filename beginning is as expected
+    std::string path_appendix = "/map_data_";
```

Comment:
[nit] appendix is usually at the end of the associated object, use prefix if you are refering to the front